azurerm/resource_arm_virtual_machine.go

```diff
@@ -721,91 +721,90 @@ func resourceArmVirtualMachineRead(d *schema.ResourceData, meta interface{}) err
-				return fmt.Errorf("[DEBUG] Error setting Virtual Machine Storage OS Profile Windows Configuration: %#v", err)
+			if dataDisks := profile.DataDisks; dataDisks != nil {
+				disksInfo := make([]*compute.Disk, len(*dataDisks))
+				for i, dataDisk := range *dataDisks {
+					diskInfo, err := resourceArmVirtualMachineGetManagedDiskInfo(dataDisk.ManagedDisk, meta)
+					if err != nil {
+						return fmt.Errorf("[DEBUG] Error getting managed data disk detailed information: %#v", err)
+					}
+					disksInfo[i] = diskInfo
+				}
+				if err := d.Set("storage_data_disk", flattenAzureRmVirtualMachineDataDisk(dataDisks, disksInfo)); err != nil {
+					return fmt.Errorf("[DEBUG] Error setting Virtual Machine Storage Data Disks error: %#v", err)
+				}
 			}
 		}
 
-		if resp.VirtualMachineProperties.OsProfile.LinuxConfiguration != nil {
-			if err := d.Set("os_profile_linux_config", schema.NewSet(resourceArmVirtualMachineStorageOsProfileLinuxConfigHash, flattenAzureRmVirtualMachineOsProfileLinuxConfiguration(resp.VirtualMachineProperties.OsProfile.LinuxConfiguration))); err != nil {
-				return fmt.Errorf("[DEBUG] Error setting Virtual Machine Storage OS Profile Linux Configuration: %#v", err)
+		if profile := props.OsProfile; profile != nil {
+			if err := d.Set("os_profile", schema.NewSet(resourceArmVirtualMachineStorageOsProfileHash, flattenAzureRmVirtualMachineOsProfile(profile))); err != nil {
+				return fmt.Errorf("Error setting `os_profile`: %#v", err)
 			}
-		}
 
-		if resp.VirtualMachineProperties.OsProfile.Secrets != nil {
-			if err := d.Set("os_profile_secrets", flattenAzureRmVirtualMachineOsProfileSecrets(resp.VirtualMachineProperties.OsProfile.Secrets)); err != nil {
-				return fmt.Errorf("[DEBUG] Error setting Virtual Machine Storage OS Profile Secrets: %#v", err)
+			if err := d.Set("os_profile_linux_config", schema.NewSet(resourceArmVirtualMachineStorageOsProfileLinuxConfigHash, flattenAzureRmVirtualMachineOsProfileLinuxConfiguration(profile.LinuxConfiguration))); err != nil {
+				return fmt.Errorf("Error setting `os_profile_linux_config`: %+v", err)
+			}
+
+			if err := d.Set("os_profile_windows_config", schema.NewSet(resourceArmVirtualMachineStorageOsProfileWindowsConfigHash, flattenAzureRmVirtualMachineOsProfileWindowsConfiguration(profile.WindowsConfiguration))); err != nil {
+				return fmt.Errorf("Error setting `os_profile_windows_config`: %+v", err)
 			}
-		}
-	}
 
-	if resp.VirtualMachineProperties.DiagnosticsProfile != nil && resp.VirtualMachineProperties.DiagnosticsProfile.BootDiagnostics != nil {
-		if err := d.Set("boot_diagnostics", flattenAzureRmVirtualMachineDiagnosticsProfile(resp.VirtualMachineProperties.DiagnosticsProfile.BootDiagnostics)); err != nil {
-			return fmt.Errorf("[DEBUG] Error setting Virtual Machine Diagnostics Profile: %#v", err)
+			if err := d.Set("os_profile_secrets", flattenAzureRmVirtualMachineOsProfileSecrets(profile.Secrets)); err != nil {
```

Comment:
Same thing here, since the code was refactored the `err` object is in the same scope and doesn't need to be re-initialized in line 777, 781, and 785, shouldn't those lines be just `if err =`?

Reply:
(as above) since we're not using this variable below it doesn't matter this variables re-defined (since it's not going to conflict / which is why the linter isn't catching this) - so whilst we could, I don't _think_ we need to here

Reply:
going to merge this either way, we can fix this up later if needed